### PR TITLE
Doc improvements

### PR DESCRIPTION
**Pull request by AArnott**

### CODE REVIEW

File: zcash_primitives/src/transaction/fees/zip317.rs [hunk 1]
```diff
@@ -44,9 +44,9 @@ pub const MINIMUM_FEE: NonNegativeAmount = NonNegativeAmount::const_from_u64(10_
 
 /// A [`FeeRule`] implementation that implements the [ZIP 317] fee rule.
 ///
-/// This fee rule supports only P2pkh transparent inputs; an error will be returned if a coin
-/// containing a non-p2pkh script is provided as an input.  This fee rule may slightly overestimate
-/// fees in case where the user is attempting to spend more than ~150 transparent inputs.
+/// This fee rule supports inputs from orchard, sapling, and transparent pools (P2PKH only).
+/// An error will be returned if a coin containing a non-p2pkh script is provided as an input.
```
Comment (str4d) [inline zcash_primitives/src/transaction/fees/zip317.rs:48]:
```suggestion
/// This fee rule supports Orchard, Sapling, and (P2PKH only) transparent inputs.
/// Returns an error if a coin containing a non-p2pkh script is provided as an input.
```

File: zcash_keys/src/encoding.rs [hunk 2]
```diff
@@ -76,13 +76,24 @@ where
     }
 }
 
+/// A trait for encoding and decoding Zcash addresses.
 pub trait AddressCodec<P>
```
Comment (str4d) [inline zcash_keys/src/encoding.rs:80]: I'm happy to merge the PR with these docs, but this trait is almost certainly going away (as the address encoding and decoding logic here is duplicative of `zcash_address`).

**Review by str4d — CHANGES_REQUESTED** [review]

**Review by nuttycom — APPROVED** [review]
ACK, doc-only.

**Review by str4d — APPROVED** [review]
ACK 296e770ec827c9a500e6751f681e1d79371734a9

**Pull request merged**